### DISCUSSION TITLE
Ask for guide to parse comments

### DISCUSSION THREAD

**Discussion by shuizaiku7** [taocpp/PEGTL#349] [Q&A] [answered]
Hi, I'm using PEGTL to parse some kind of language,
and i wanna parse the comments now,
the comments are just like the comments in C++, where '//' means comments of the line, '/* */' means comments for multiple lines.
But i don't know how to do this using PEGTL.
Could anyone give me any advice?

**Comment by ColinH** [comment, marked as answer]
Many grammars allow comments in the same places they allow spaces and tabs. So one common way to handle comments is to make a rule that accepts spaces, tabs and comments.

For an example see the `ws` rule in our taoCONFIG library at https://github.com/taocpp/json/blob/main/include/tao/json/jaxn/internal/grammar.hpp#L32 .

It allows for space, tab, cr and lf as "whitespace" characters, and for three kinds of comments, `#` and until end-of-line, `//` and until end-of-line, and non-nesting `/*` until `*/`.

**Reply by shuizaiku7** [reply]
yeah, thanks for your prompt.
i just write rule for oneLineComment and multiLineComment,
and then combine them with space and maybeSpace, and it works.
